### PR TITLE
[MM-20553] Convert user setting to typescript

### PR DESCRIPTION
**Pull request by hiendinhngoc**
#### Summary
Migrate 'components/user_settings' module and associated tests to TypeScript

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/13688

  JIRA: https://mattermost.atlassian.net/browse/MM-20553

### CODE REVIEW

File: components/user_settings/import_theme_modal.tsx [hunk 1]
```diff
@@ -48,7 +55,7 @@ export default class ImportThemeModal extends React.Component {
             this.setState({
                 inputError: (
                     <FormattedMessage
-                        id='user.settings.import_theme.submitError'
+                        id='user.settings.TOGGLE_IMPORT_THEME_MODAL.submitError'
```
Comment (devinbinnie) [inline components/user_settings/import_theme_modal.tsx]: Why was this translation changed?

File: components/user_settings/import_theme_modal.tsx [hunk 2]
```diff
@@ -59,26 +66,26 @@ export default class ImportThemeModal extends React.Component {
...
-        theme.sidebarTextActiveBorder = colors[2];
-        theme.sidebarTextActiveColor = colors[3];
-        theme.sidebarHeaderBg = colors[1];
-        theme.sidebarHeaderTextColor = colors[5];
-        theme.onlineIndicator = colors[6];
-        theme.mentionBg = colors[7];
-        setThemeDefaults(theme);
-
-        this.state.callback(theme);
+        (theme as any).sidebarBg = colors[0];
```
Comment (devinbinnie) [inline components/user_settings/import_theme_modal.tsx]: We have a `Theme` type you can use instead, it's in `mattermost-redux`.

Comment (hiendinhngoc) [reply components/user_settings/import_theme_modal.tsx]: `import {Theme} from 'mattermost-redux'`
gave me error: `Module '"../../node_modules/mattermost-redux"' has no exported member 'Theme'.`
what was I wrong here?

Comment (devinbinnie) [reply components/user_settings/import_theme_modal.tsx]: You need to define the path that the `Theme` type is in.
It should look like:
`import {Theme} from 'mattermost-redux/types/preferences'`

Comment (hiendinhngoc) [reply components/user_settings/import_theme_modal.tsx]: thank you!

**Pull request merged**